components/base/wheeled/wheeled_base_test.go

```diff
@@ -353,3 +357,252 @@ func TestValidate(t *testing.T) {
+		measYaw, err := getCurrentYaw(ctx, ms, extra)
+		test.That(t, measYaw, test.ShouldEqual, calcYaw)
+		test.That(t, measYaw > 0, test.ShouldBeTrue)
+		test.That(t, calcYaw > 0, test.ShouldBeTrue)
+		test.That(t, err, test.ShouldBeNil)
+	}
+
+	// test findSpin Parameters calculations
+	for _, params := range []struct {
+		added float64
+		speed float64
+		start float64
+		dir   float64
+		goal  float64
+	}{
+		{-15, -20, 0, 1, 0},
+		{20, -20, 360, -1, 0},
+		{90, 10, 10, 1, 0},
+	} {
+		params.goal = addAnglesInDomain(params.start, params.added)
+		goal, dir := findSpinParams(params.added, params.speed, params.start)
+		test.That(t, goal, test.ShouldAlmostEqual, params.goal)
+		test.That(t, dir, test.ShouldAlmostEqual, params.dir)
+	}
+}
+
+func TestHasOverShot(t *testing.T) {
+	dirCases := []dirInfo{
+		{"ccw", 1.0},
+		{"cw", -1},
+	}
+
+	addCases := []addInfo{
+		{"acute", 3},
+		{"acute", 20},
+		{"right", 90},
+		{"obtuse", 110},
+		{"straight", 180},
+		{"reflex", 200},
+		{"reflexright", 270},
+		{"reflexplus", 345},
+		{"complete", 357},
+	}
+
+	startCases := []float64{
+		// TODO RSDK- refine overshot cases, add 3 around 360 range
+		5,
+		12,
+		15,
+		45,
+		90,
+		135,
+		180,
+		225,
+		270,
+		315,
+		260,
+		270,
+		345,
+		355,
+		359,
+	}
+
+	/*
+			definition of quadrants and directions
+			q2	 	|		q1	  <-| ccw (+ve)
+				+ve	|  +ve			|
+		180 ________|________ 0
+					|
+				-ve	|  -ve			|
+			q3		|		q4	  <-| cw (-ve)
+	*/
+
+	for _, dirCase := range dirCases {
+		for _, addCase := range addCases {
+			for _, start := range startCases {
+				condition := makeCondition(addCase, dirCase, start)
+
+				start := condition.Start
+				target := condition.Target
+				dir := condition.Direction
+				added := addCase.Value
+
+				// test a few cases in range ensure were not falsely overshooting
+				notovers := map[string]float64{
+					"start":    addAnglesInDomain(start, 0),
+					"under:+":  addAnglesInDomain(start, dir),
+					"under:++": addAnglesInDomain(start, dir*added/2),
+					"under:--": addAnglesInDomain(target, -dir*added/2),
+					"under:-":  addAnglesInDomain(target, -dir),
+					"end:":     addAnglesInDomain(target, 0),
+					// TODO: RSDK- refine overshot cases, test end and cw failure
+					"over:": addAnglesInDomain(target, dir),
+				}
+				for key, angle := range notovers {
+					noStr := "[" + strconv.FormatFloat(angle, 'f', 1, 64) + "]"
+					t.Run(condition.Name+noStr+key, func(t *testing.T) {
+						if key == "end:" || key == "over:" {
+							// skipped edge case
+							if key == "end:" && dirCase.Name == "cw" && target == 0.0 {
+								t.Skip()
+							} else {
+								test.That(t,
+									hasOverShot(angle, start, target, dir),
+									test.ShouldBeTrue)
+							}
+						} else {
+							test.That(t,
+								hasOverShot(angle, start, target, dir),
+								test.ShouldBeFalse)
+						}
+					})
+				}
+			}
+		}
+	}
+}
+
+type TestCase struct {
```

Comment:
the problem with having this much infrastructure and logic in a test file is that there are now two reasons a test can be failing: the actual code _and_ the test infrastructure. I think it would be better to create these names non programmatically even though it may result in a longer file

Reply:
do you mean just shove these definitions into anonymous structs into thee test file?